cpp/src/parquet/metadata.h

```diff
@@ -182,6 +182,28 @@ class PARQUET_EXPORT ColumnChunkMetaData {
   std::unique_ptr<ColumnChunkMetaDataImpl> impl_;
 };
 
+/// \brief DataPageStats is a proxy around stats in format::PageHeader.
+class PARQUET_EXPORT DataPageStats {
+ public:
+  static std::unique_ptr<DataPageStats> Make(const void* page_header);
+
+  ~DataPageStats();
+
+  bool Equals(const DataPageStats& other) const;
+
+  int32_t num_values() const;
```

Comment:
There is already a notion of statistics (see statistics.h), is there a reason to not use that as a member here?

Reply:
The statistics has more functionality/dependency than we need. It is meant for accumulating values and computing the min/max. Also, we need a column descriptor to make it.

Reply:
EncodedStatistics looks almost identical to this though?  For normal statistics It looks the like the descriptor is only needed to construct a comparator, I wonder if there is some refactoring that could separate the two?  It seems like having a 3rd type of statitistics would add confusion, so I want to make sure there is a strong reason to add it.

Reply:
Yes, we could use EncodedStatistics, it is missing num_rows and num_values though. Should we refactor EncodedStatistics to meet our needs?

Reply:
I have a slightly preference for composing encoded stats into this new object, rather than refactoring.  Do you have a preference?

Reply:
I changed this part to use encoded stats. Please take a look.